### PR TITLE
Homework15

### PR DESCRIPTION
**Pull request by bodenkovv**
### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title
2. ...

### Manual testing scenarios

1. ...
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/BodenkoVV/AskQuestion/Block/TabAskQuestion.php [hunk 1]
```diff
@@ -24,29 +24,47 @@ class TabAskQuestion extends Template
...
-        Data $helperData,
+//        Data $helperData,
         Context $context,
         \Magento\Framework\Registry $registry,
+        \Magento\Framework\View\Result\PageFactory $resultPageFactory,
         array $data = []
     ) {
         parent::__construct($context, $data);
-        $this->helperData = $helperData;
+//        $this->helperData = $helperData;
```
Comment (chabanenk0) [inline app/code/BodenkoVV/AskQuestion/Block/TabAskQuestion.php]: don't comment the code which appears not needed. Remove it.

File: app/code/BodenkoVV/AskQuestion/Block/TabAskQuestion.php [hunk 2]
```diff
@@ -24,29 +24,47 @@ class TabAskQuestion extends Template
...
         $this->_registry = $registry;
+        $this->resultPageFactory = $resultPageFactory;
+    }
+
+    /**
+     * @return \Magento\Framework\View\Result\Page
+     */
+    public function execute()
+    {
+//        $resultPage = $this->resultPageFactory->create();
```
Comment (chabanenk0) [inline app/code/BodenkoVV/AskQuestion/Block/TabAskQuestion.php]: remove if not needed, or uncomment and fix if needed

File: app/code/BodenkoVV/AskQuestion/Block/TabAskQuestion.php [hunk 3]
```diff
@@ -2,57 +2,113 @@
...
     {
-        return parent::_prepareLayout();
+        $this->getQuestionConfig();
+        parent::_toHtml();
+    }
+
+    public function getQuestionConfig()
+    {
+        $resultPage = $this->resultPageFactory->create();
+        if ($this->helperData->getGeneralConfig('bodenkovv_askquestion_enable'))
```
Comment (chabanenk0) [inline app/code/BodenkoVV/AskQuestion/Block/TabAskQuestion.php]: This check is not correct. The field bodenkovv_askquestion_enable is not in general section, it's inside of 'cron' section
Please rename this item to "enable_cron" or 'enable_auto_confirming', no need to add vendor name there.
Also I would advice not to use getGeneralConfig, please create more simple function in helper: isCronEnabled() and use the exact constant 'askquestion/cron/enable_auto_confirming'

Comment (bodenkovv) [reply app/code/BodenkoVV/AskQuestion/Block/TabAskQuestion.php]: bodenkovv_askquestion_enable in the general section )

Comment (bodenkovv) [reply app/code/BodenkoVV/AskQuestion/Block/TabAskQuestion.php]: +

**Pull request merged**